### PR TITLE
further updates to 963

### PR DESCRIPTION
**Pull request by lucasnz**
Layout tweaks.
I note some work still required with filter screen.

### CODE REVIEW

**Comment by pliablepixels** [comment]
So far, this PR leads to more whitespace at the bottom while master seems to be a tighter fit.
What issue are you seeing with master?

**Comment by lucasnz** [comment]
> 
> 
> So far, this PR leads to more whitespace at the bottom while master seems to be a tighter fit.
> What issue are you seeing with master?

Are you talking about whitespace in the large or small thumbs view. The intent of this pull request is to get the large thumbs view to look better while leaving the smalls thumbs the same. This is what I'm seeing, where is the whitespace you are referring to?

Screen shoots from browser...

Small thumbs. Before left, after right:
image

Large thumbs. Before left, after right:
image

**Comment by lucasnz** [comment]
I still have a problem with resizing (landscape to portrait) correctly when in large thumbs. Not sure how to resolve this. The code looks right, but it doesn't resize the collection repeat objects (it does change the images).

**Comment by pliablepixels** [comment]
check out latest master - should fix the extra space.
When run on a device, this PR scales the image incorrectly:

IMG_0351

**Comment by pliablepixels** [comment]
This exception occurs when refresh conflicts with infinite scroll. You can trigger it by pulling down to refresh as well as tapping on 3 dots->refresh. It will also occur is you are at the top and auto refresh is on.

This entire event loading code is such a huge mess, its hard to follow what I did

**Comment by pliablepixels** [comment]
Okay, I reset the code in getInitialEvents and loadMore to the old code. I've kept the maxEventsToLoad variable instead of hard-coding to 50 so you get your object detect gif optimization. Also have kept the delta time check code. This took the exception away. There was something in the new code that was triggering this.

I've also removed pull to refresh (it was off earlier) - the reason I took it away earlier was I found large lists and pull to refresh always caused interaction issues - as you scroll fast, especially on a device, it is very easy to trigger a refresh by mistake. Instead, the user can choose to refresh via the menu.

Overall, this code is decidedly terrible and needs to be cleaned up. There are a lot of useless variables and unnecessary specialized checks that needs to relooked at some day. For now, this was a blocking problem that was stopping me from making a new release.

**Comment by lucasnz** [comment]
The screenshots above were from your master. I've just pulled the latest again. Still lots of whitespace;
image

I notice that scaling issue sometimes on my device, so clearly, this PR isn't quite right. I think we're getting closer to the right set up. The key things I'd like to achieve are;
1. Remove unneeded whitespace (this problem seems to effect both small and large thumbs for me).
2. Remove the br between time and event name (large thumbs only) - this can be replaced with a span with the css I used
3. Move the information line to below the image (large thumbs only)

There's clearly an issue with this PR. I have 1 minor commit that addresses 2 & 3. So, I think I'll cancel this PR and create a new one (covering only 2&3 - they're simple changes). Then, I still think we need to work on the whitespace....

**Comment by lucasnz** [comment]
Ah - my Android has $rootScope.devWidth = 1080. That explains the large thumbs cutting off the screen...

**Comment by pliablepixels** [comment]
1. Does the white space go away if you rotate the phone and back?
2. What happens if you remove the multiplication with pixelRatio in NVR.computeDeviceSize?

**Comment by lucasnz** [comment]
1. Yes, I tried rotating and is does look better...
2. I already tried removing the multiplier and returns the same value.
Googling has landed me on this, and I'm currently playing...
https://ionicframework.com/docs/angular/platform#islandscape-boolean

**Comment by lucasnz** [comment]
FYI: debug log shows: resize/orient: 1080(w) * 2160(h). But that must be the native resolution and not the scaled resolution.

**Comment by pliablepixels** [comment]
So that is what we need to solve. If the width/height computed is not accurate, the whitespaces will occur.

1. What happens if you wrap the computeDeviceSize code in a 300 timeout in NVR.js? That is, lets give enough time for the view to draw before it computes

2. What should the actual size be? If you know it, as a test, just hard code it in commputeDeviceSize. Does it resolve?

**Comment by lucasnz** [comment]
app.js already has a delay. Should I just bump the number there?
      $timeout (function () {

        NVR.computeDeviceSize();

      },30);

**Comment by lucasnz** [comment]
wait, I see pixelratio is 3. Maybe I didn't test it correctly the first time I tried removing this multiplier...

**Comment by lucasnz** [comment]
Right, we are getting somewhere. If I remark out these, I get what looks like the correct height/widths;
        //$rootScope.devWidth *= pixelRatio;
        //$rootScope.devHeight *= pixelRatio;
But I still have way too much whitespace. But that doesn't surprise me since I see lots of whitespace in the browser too (and the browser didn't have the scaling issue).... But we're getting closer to solving it.

**Comment by lucasnz** [comment]
ok, just done some more testing, and I'm pretty confident we need to remark out pixel ratio multiplier. I'm not sure how this effects other OS's (was that incorrectly scaled screenshot you sent me on Android or IOS?). I suspect it needs to be remarked out for all...

I still have way too much whitespace (see earlier screenshots - no change really). As far as I can tell, the problem is the size of list item is incorrectly calculated. I think we should base the list item height, on the image height directly. Something vaguely like this;
https://github.com/pliablepixels/zmNinja/pull/970/commits/187bbf5f78824069ca4b1937b5a227745937719a

I really need to go to bed. But I'm satisfied I've moved things forward somewhat. Hope that helps.

**Comment by pliablepixels** [comment]
Yup, I'll comment out pixelRatio as well after testing with iOS. It was off earlier as well.
In looking at your images again, I see that whitespace occurs only when the image cannot be scaled any further without affecting aspect ratio. This comes from the fact that we have a fixed row height (and we can't make it dynamic). IF you have a mix of different camera heights, you'll see some with whitespace and some without. Basically, the longer the height of the window is stretched, the more whitespace you'll see between a horizontal image (w>h) and the row.

**Comment by lucasnz** [comment]
Yes, I thought about having different camera heights not being handled in that commit, but I think it can be handled. Just need to store the regular height for each event as meta-data on the event... I found myself solving this problem as I was falling to sleep last night - haha. That commit of mine (https://github.com/pliablepixels/zmNinja/commit/187bbf5f78824069ca4b1937b5a227745937719a) in hindsight was a bit crap. I might have a go at a better one tomorrow. As it also effects the navBar header, and I had an idea on how to solve this too... But I really do need to hit the hay.
